src/generator/models.ts

```diff
@@ -432,10 +427,8 @@ function generateStruct(lang: Language, props?: Property[]): StructDef {
   return st;
 }
 
-function generateOptionalParamsStruct(optionalParam: Parameter): StructDef {
-  // for the optional params struct, the params are smuggled via GroupProperty
-  const params = (<GroupProperty>(<ObjectSchema>optionalParam.schema).properties![0]).originalParameter;
-  const st = new StructDef(optionalParam.schema.language.go!, undefined, params);
```

Comment:
Are parameter groups always optional? Since I'm seeing that it's replacing this.

Reply:
Parameter groups are specified with [x-ms-parameter-grouping](https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-parameter-grouping) which is optional.

The creation of the optional params struct was reworked to use this same mechanism, see link 258 in transform.ts.